internal/datanode/segment.go

```diff
@@ -162,23 +69,32 @@ func (s *Segment) setType(t datapb.SegmentType) {
 	s.sType.Store(t)
 }
 
-func (s *Segment) updatePKRange(ids storage.FieldData) error {
-	log := log.With(zap.Int64("collectionID", s.collectionID),
-		zap.Int64("partitionID", s.partitionID),
-		zap.Int64("segmentID", s.segmentID),
-	)
-
-	err := s.pkStat.updatePKRange(ids)
+func (s *Segment) updatePKRange(ids storage.FieldData) {
+	s.InitCurrentStat()
+	err := s.currentStat.UpdatePKRange(ids)
 	if err != nil {
-		log.Warn("failed to updatePKRange", zap.Error(err))
+		panic(err)
 	}
+}
 
-	log.Info("update pk range",
-		zap.Int64("num_rows", s.numRows), zap.Any("minPK", s.pkStat.minPK), zap.Any("maxPK", s.pkStat.maxPK))
-
-	return nil
+func (s *Segment) InitCurrentStat() {
+	if s.currentStat == nil {
+		s.currentStat = &storage.PkStatistics{
+			PkFilter: bloom.NewWithEstimates(storage.BloomFilterSize, storage.MaxBloomFalsePositive),
```

Comment:
NIT: `storage.BloomFilterSize` is a magic number. Maybe estimation of `buffer size / row size` is better here?

Reply:
> 

Because it's simply a temporary buffer, so I guess doesn't really matter